### DISCUSSION TITLE
use numpad keys to trigger apps

### DISCUSSION THREAD

**Discussion by stuart-leitch** [rxhanson/Charmstone-Community#23] [Ideas]
Charmstone is great, and I'm building the muscle memory for where my main apps are.  One thing is that when my hands are on the keyboard, it's a (very minor) jump to the mouse.

Could charmstone be displayed when its modifier keys are pressed,  and then allow the number pad keys to trigger the equivalent app (/script)?

Feels like the flip side of the option to have a mouse button as the trigger (i.e. all mouse or all kb).

**Comment by rxhanson** [comment]
Thanks for the feedback.

> Could charmstone be displayed when its modifier keys are pressed, and then allow the number pad keys to trigger the equivalent app (/script)?

My plan for the next feature release of Charmstone was to add shortcut recorders for each of the positions, which you could configure with the num pad if you desired, but maybe it would be nice to just have a checkbox for using the configured modifiers with the numpad numbers, similar to the arrow keys checkbox.

**Reply by stuart-leitch** [reply]
Thanks,  that sounds great.  I've also just discovered the use of the arrow keys which gives mouse-free operation.

One thing I think I'd like (but there may be a good reason for not having!) is to have charmstone display when the modifiers are first pressed, rather than waiting for the mouse to move). I find while I'm waiting to build the muscle memory of what is where I'd like the icons straight away.

Thanks for a great product.  Between this and rectangle pro you've really made a big difference to how I use my superwide display.

**Reply by rxhanson** [reply]
> One thing I think I'd like (but there may be a good reason for not having!) is to have charmstone display when the modifiers are first pressed, rather than waiting for the mouse to move). I find while I'm waiting to build the muscle memory of what is where I'd like the icons straight away.

I didn't mention it, but you can get this by sliding the "Min Dist to Appear" all the way to the left in the preferences.

**Reply by henryliuser** [reply]
I'd also really appreciate it if you could add hjkl (left down up right) support to be in line with Vim users. Of course the shortcut recorder would work fine for this as well. Thanks for an awesome app!

**Reply by darpham** [reply]
@rxhanson Any updates on custom shortcut recorders, I'd also like to re-map to hjkl for vim motions similar to @henryliuser 's request. Thanks! :)

**Reply by rxhanson** [reply]
Unfortunately no updates here at the moment. I'll put another +1 on this one for when I'm back on Charmstone feature work again.

**Reply by aaronmw** [reply]
Impatient as I am, I used BetterTouchTool (not free) to painstakingly map each numpad key to launch the app in Charmstone's corresponding position. I think you could probably achieve the same with a more purpose-built keyboard-mapping tool like Karabiner (free! https://karabiner-elements.pqrs.org/index.html) or similar 👍

More tools: https://www.makeuseof.com/tag/apps-customize-mac-keyboard/
